app-frontend/src/app/components/pages/project/analyses/analyses/index.js

```diff
@@ -194,6 +202,92 @@ class AnalysesListController {
             });
         });
     }
+
+    onToggle(id) {
+        if (this.selected.has(id)) {
+            this.selected = this.selected.delete(id);
+        } else {
+            const analysis = this.itemList.find(a => a.id === id);
+            this.selected = this.selected.set(id, analysis);
+        }
+        this.updateSelectText();
+    }
+
+    allVisibleSelected() {
+        // const templateId = _.get(this.selected.values().next(), 'value.templateId');
+        let itemSet = new Set(
+            this.itemList
+                // .filter(i => i.templateId === templateId)
+                .map(l => l.id));
+        return this.selected.size &&
+            itemSet.intersect(this.selected.keySeq()).size === itemSet.size;
+    }
+
+    isSelectable(item) {
+        if (this.selected.size) {
+            const templateId = _.get(this.selected.values().next(), 'value.templateId');
+            return item.templateId === templateId;
+        }
+        return true;
+    }
+
+    canEditSelection() {
+        const templateIds = new Set(this.selected.valueSeq().map(v => v.templateId));
+        return templateIds.size <= 1;
+    }
+
+    selectAll() {
+        if (this.allVisibleSelected()) {
+            this.selected = this.selected.clear();
+        } else {
+            this.selected = this.selected.merge(
+                new Map(
+                    this.itemList
+                        .map(i => [i.id, i]))
+            );
+        }
+        this.updateSelectText();
+    }
+
+    selectGroup(analysis) {
+        const templateId = analysis.templateId;
+        this.selected = this.selected.merge(
```

Comment:
Also, under multiple selection mode, if you click on one of the analyses' actions dropdown and click `Edit` or `Delete`, it will do such action for that analysis only, which is correct, since it is initiated from an individual analysis, but the multiple selections are still on, which is a bit confusing. So I propose two potential solutions:
1. Disable actions dropdown under multiple selection mode
2. Update the selected `Map` so that it only contains the analysis the action is initiated from (I personally prefer this one though)

Reply:
yea it is a bit confusing. I'll think about it for a bit. @designmatty any suggestions?

Reply:
OS's tend to handle this by not having "menu" buttons, and instead having contextual dropdowns when you right click things which apply to everything selected (or the single item you right clicked if none selected)

Reply:
Added a count of analyses being edited to the edit modal per conversation with @designmatty